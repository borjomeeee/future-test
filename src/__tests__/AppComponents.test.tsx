import React from "react";

import { render } from "@testing-library/react";

import AppContext from "../context/App.context";

import TableComponent from "../components/Table.component";
import { ITableItem } from "../models/TableItem.model";

const tableItem: ITableItem = {
  id: 1,
  firstName: "Имя",
  lastName: "Фамилия",
  email: "Почта",
  phone: "Телефон",

  address: {
    streetAddress: "Улица",
    city: "Город",
    state: "Штат",
    zip: "Индекс",
  },
  description: "Описание",
};

it("table render correctly", () => {
  render(
    <AppContext.Provider
      value={{
        isLoading: false,
        error: "",

        tableItems: [tableItem, tableItem, tableItem, tableItem, tableItem],

        dispatch: () => undefined,
      }}
    >
      <TableComponent />
    </AppContext.Provider>
  );
  expect(document.getElementsByClassName("table")[0]).toMatchSnapshot();
});

it("table form render correctly", () => {
  render(
    <AppContext.Provider
      value={{
        isLoading: false,
        error: "",

        tableItems: [],
        dispatch: () => undefined,
      }}
    >
      <TableComponent />
    </AppContext.Provider>
  );
  expect(document.getElementsByClassName("table__form")[0]).toMatchSnapshot();
});
