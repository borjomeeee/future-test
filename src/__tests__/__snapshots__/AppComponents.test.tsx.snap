// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<tr
  class="table__head"
/>
`;

exports[`table form render correctly 1`] = `undefined`;

exports[`table render correctly 1`] = `
<div
  class="table"
>
  <div
    class="table__top-bar"
  >
    <button
      class="top-bar__add-button btn btn-primary"
      type="button"
    >
      Добавить
    </button>
    <div
      class="top-bar__search-container search-container"
    >
      <div
        class="search-container__container"
      >
        <div
          class="search-container__input-container"
        >
          <input
            class="search-container__input form-control"
            id="table-search"
            placeholder="Введите хоть что-то..."
            type="text"
          />
        </div>
        <div
          class="search-container__button-container"
        >
          <button
            class="search-container__button btn btn-success"
            type="button"
          >
            Найти
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="table__container"
  >
    <table
      class="table__content "
    >
      <thead>
        <tr
          class="table__head"
        />
      </thead>
      <tbody>
        <tr
          class="table__col "
        >
          <th
            scope="row"
          >
            1
          </th>
          <td>
            Имя
          </td>
          <td>
            Фамилия
          </td>
          <td>
            Почта
          </td>
          <td>
            Телефон
          </td>
        </tr>
        <tr
          class="table__col "
        >
          <th
            scope="row"
          >
            1
          </th>
          <td>
            Имя
          </td>
          <td>
            Фамилия
          </td>
          <td>
            Почта
          </td>
          <td>
            Телефон
          </td>
        </tr>
        <tr
          class="table__col "
        >
          <th
            scope="row"
          >
            1
          </th>
          <td>
            Имя
          </td>
          <td>
            Фамилия
          </td>
          <td>
            Почта
          </td>
          <td>
            Телефон
          </td>
        </tr>
        <tr
          class="table__col "
        >
          <th
            scope="row"
          >
            1
          </th>
          <td>
            Имя
          </td>
          <td>
            Фамилия
          </td>
          <td>
            Почта
          </td>
          <td>
            Телефон
          </td>
        </tr>
        <tr
          class="table__col "
        >
          <th
            scope="row"
          >
            1
          </th>
          <td>
            Имя
          </td>
          <td>
            Фамилия
          </td>
          <td>
            Почта
          </td>
          <td>
            Телефон
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    class="table__pagination pagination"
  >
    <ul
      class="pagination__container "
    >
      <li
        class="pagination__button pagination__button_disabled"
      >
        <a
          aria-disabled="true"
          role="button"
          tabindex="0"
        >
          Назад
        </a>
      </li>
      <li
        class="pagination__page-link pagination__page-link_active"
      >
        <a
          aria-current="page"
          aria-label="Page 1 is your current page"
          role="button"
          tabindex="0"
        >
          1
        </a>
      </li>
      <li
        class="pagination__button pagination__button_disabled"
      >
        <a
          aria-disabled="true"
          role="button"
          tabindex="0"
        >
          Вперед
        </a>
      </li>
    </ul>
  </div>
</div>
`;
